lib/rails_cursor_pagination/paginator.rb

```diff
@@ -432,8 +409,25 @@ def relation_with_cursor_fields
     #
     # @return [ActiveRecord::Relation]
     def sorted_relation
+      unless custom_order_field?
+        return relation_with_cursor_fields.reorder id: pagination_sorting.upcase
+      end
+
       relation_with_cursor_fields
-        .reorder(sql_column => pagination_sorting.upcase)
+        .reorder(@order_field => pagination_sorting.upcase,
+                 id: pagination_sorting.upcase)
+    end
+
+    # Return a properly escaped reference to the ID column prefixed with the
+    # table name. This prefixing is important in case of another model having
+    # been joined to the passed relation.
+    #
+    # @return [String (frozen)]
```

Comment:
Since you say `frozen`, should you also `freeze` the string before returning it?

Reply:
This file has the magic comment 
https://github.com/xing/rails_cursor_pagination/blob/f107a6506659f63245fee7084e21cbd14e327dec/lib/rails_cursor_pagination/paginator.rb#L1
at the very top of the class definition. Therefore, all strings that are defined in it are automatically frozen. At least this is the case _in Ruby 2.7_ which I used for local development. And therefore my IDE inferred that this is returning a frozen string.

However, I just did some more digging and [found that in Ruby 3.0 _interpolated strings_ will no longer be automatically frozen](https://bugs.ruby-lang.org/issues/17104) (as they can potentially contain very dynamic content. In our case, it can fairly safely be assumed that there is a limited sub-set of strings that will be seen here, as there are only so many models and columns that will be used for sorting. Therefore, having it as a frozen string actually makes sense from a performance perspective.

So yes, for Ruby 3.0's sake I should add a `#freeze` call here 👍  Thanks for flagging!

Reply:
Done – but it also required disabling the `Style/RedundantFreeze` rubocop check since Rubocop is checking for compliance with Ruby 2.5 syntax (where the string is auto-frozen).

```
lib/rails_cursor_pagination/paginator.rb:430:7: C: [Correctable] Style/RedundantFreeze: Do not freeze immutable objects, as freezing them has no effect.
      "#{escaped_table_name}.#{escaped_id_column}".freeze
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```